WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -470,31 +470,40 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
 
         switch section {
         case .siteMenu:
-            blogDetailsViewController?.pulledToRefresh(with: refreshControl) { [weak self] in
-                guard let self = self else {
-                    return
-                }
 
-                self.updateNavigationTitle(for: blog)
-                self.sitePickerViewController?.blogDetailHeaderView.blog = blog
-                self.updateBlazeStatus(for: blog)
+            let pulledToRefresh: () -> Void = {
+                self.blogDetailsViewController?.pulledToRefresh(with: self.refreshControl) { [weak self] in
+                    guard let self = self else {
+                        return
+                    }
+
+                    self.updateNavigationTitle(for: blog)
+                    self.sitePickerViewController?.blogDetailHeaderView.blog = blog
+                    self.updateBlazeStatusAndReloadChildViewController(for: blog)
```

Comment:
Right now, when the menu is refreshed, we call the blaze endpoint twice. That's not very optimum.

Reply:
Oops! Line 482 isn't meant to be there - but I followed your suggestion and refactored the code to handle fetching the blaze status inside `BlogService.syncBlogAndAllMetadata()`

[2ff0a02](https://github.com/wordpress-mobile/WordPress-iOS/pull/20290/commits/2ff0a02cb25a51dd9f0a99e0e409f01b2859eae1)